### PR TITLE
feat: dump heap prof

### PR DESCRIPTION
**Pull request by yuhao-su**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

- dump memory profile when the memory usage exceed a threshold

Configuration:
```toml
[streaming.auto_dump_heap_profile]
dir = "/your_profile_dir"
threshold = 0.9
```

The `dir` can be overwritten by setting env `RW_AUTO_DUMP_HEAP_PROFILE_DIR`.

When the `dir` is set to something other than empty, the heap profile will be dumped to `dir`

`streaming.auto_dump_heap_profile.dir` should only be set when `MALLOC_CONF="prof:true"`

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR contains user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

**Comment by fuyufjh** [comment]
Please fill the PR description

File: src/config/example.toml [hunk 1]
```diff
@@ -61,6 +61,10 @@ in_flight_barrier_nums = 10000
 async_stack_trace = "ReleaseVerbose"
 unique_user_stream_errors = 10
 
+[streaming.auto_dump_heap_profile]
```
Comment (fuyufjh) [inline src/config/example.toml:64]: Emmm I think it's more suitable to be placed on `[server]` section

File: src/tests/simulation/Cargo.toml [hunk 2]
```diff
@@ -43,6 +43,7 @@ serde_derive = "1.0.152"
 serde_json = "1.0.103"
 sqllogictest = "0.15.1"
 tempfile = "3"
+tikv-jemallocator = { git = "https://github.com/yuhao-su/jemallocator.git", branch = "yuhao/prof", features = ["profiling"] }
```
Comment (fuyufjh) [inline src/tests/simulation/Cargo.toml:46]: This should only be a provisional solution. Can you please merge it into upstream (tikv/jemallocator)?

Comment (yuhao-su) [reply src/tests/simulation/Cargo.toml:46]: Sure, should I merge this pr after I merger jenallocator?

Comment (fuyufjh) [reply src/tests/simulation/Cargo.toml:46]: Not necessary I think, but make sure to fix it later ☺️

**Review by fuyufjh — APPROVED** [review]

File: src/config/example.toml (hunk 1 above)
Comment (yuhao-su) [reply src/config/example.toml:64]: Hmmm, one problem is we might want set different profile configs for cn/meta/compactor. We might need config for different servers like `[compute]/[compactor]`?

**Pull request merged**